src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -563,7 +563,7 @@ private static boolean isPrecededByJavadoc(DetailAST token) {
         boolean result = false;
         final DetailAST previous = token.getPreviousSibling();
         if (previous.getType() == TokenTypes.BLOCK_COMMENT_BEGIN
-                && JavadocUtils.isJavadocComment(previous)) {
+                && JavadocUtils.isJavadocComment(previous.getFirstChild().getText())) {
```

Comment:
java.lang.AssertionError: unexpected output: F:\CS\workspaces\Git\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\whitespace\emptylineseparator\package-info\test2\package-info.java:2: 'package' should be separated from previous statement. 
    Expected :0
    Actual   :1

The aforementioned UT was apparently passing on the basis of [`isPrecededByJavadoc`](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java#L562) which used this version of the overloaded function [`isJavadocComment`](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java#L264) which in turn called this version of [`isJavadocComment`](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java#L246). Both these versions were previously separate from [`isCorrectJavadocPosition`](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java#L305).